subsys/bluetooth/host/conn.c

```diff
@@ -339,11 +339,26 @@ void bt_conn_reset_rx_state(struct bt_conn *conn)
 	conn->rx = NULL;
 }
 
-static void bt_acl_recv(struct bt_conn *conn, struct net_buf *buf,
-			uint8_t flags)
+static void set_acl_flags(struct net_buf *buf, uint8_t value)
+{
+	if (!IS_ENABLED(CONFIG_BT_HCI_ACL_FLOW_CONTROL)) {
+		ARG_UNUSED(buf);
+		ARG_UNUSED(value);
+	}
+
+	enum acl_data_flags *buf_flags = bt_get_acl_buf_flags(buf);
+
+	__ASSERT_NO_MSG(buf->user_data_size >= sizeof(struct acl_data));
+
+	*buf_flags = value;
+}
+
+static void bt_acl_recv(struct bt_conn *conn, struct net_buf *buf, uint8_t flags)
 {
 	uint16_t acl_total_len;
 
+	set_acl_flags(buf, 0);
```

Comment:
Sometimes, but not always, we have a `NONE` flag so that we can use it like this
```suggestion
	set_acl_flags(buf, ACL_DATA_NONE);
```

This is especially useful if the function, set_acl_flags, took the `enum acl_data_flags` as as the type, instead of `uint8_t`